core/shared/src/main/scala/zio/Fiber.scala

```diff
@@ -267,17 +267,20 @@ trait Fiber[+E, +A] { self =>
    * @param f function to the error into a Throwable
    * @return `UIO[Future[A]]`
    */
-  final def toFutureWith(f: E => Throwable): UIO[Future[A]] =
+  final def toFutureWith(f: E => Throwable): UIO[CancelableFuture[E, A]] =
     UIO.effectTotal {
       val p: concurrent.Promise[A] = scala.concurrent.Promise[A]()
 
       def failure(cause: Cause[E]): UIO[p.type] = UIO(p.failure(cause.squashWith(f)))
       def success(value: A): UIO[p.type]        = UIO(p.success(value))
 
-      UIO.effectTotal(p.future) <* self.await
+      ZIO.runtime[Any].map { runtime =>
+        new CancelableFuture[E, A](p.future) {
+          def cancel: Future[Exit[E, A]] = runtime.unsafeRunToFuture(interrupt)
+        }
+      } <* self.await
         .flatMap[Any, Nothing, p.type](_.foldM[Any, Nothing, p.type](failure, success))
```

Comment:
I wonder if type annotations could be avoided if `concurrent.Promise[A]` is used in `success/failure` returns instead of `p.type`